Services.Test/SmartDictionaryTest.cs

```diff
@@ -0,0 +1,264 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Models;
+using Services.Test.helpers;
+using Xunit;
+using Xunit.Abstractions;
+using static Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Models.DeviceModel;
+
+namespace Services.Test
+{
+    public class SmartDictionaryTest
+    {
+        private ISmartDictionary target;
+
+        public SmartDictionaryTest(ITestOutputHelper log)
+        {
+            // Initialize device properties
+            this.target = this.GetTestProperties();
+        }
+
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void Should_Be_Empty_On_Get_All_When_No_Properties_Added()
```

Comment:
The naming convention can be improved. So when following `Should_ExpectedBehavior_When_StateUnderTest`. It makes sense to have `Should_BeEmpty_When_GetAllEmpty`